components/infobox/commons/infobox_character.lua

```diff
@@ -0,0 +1,123 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Infobox/Character
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local Namespace = require('Module:Namespace')
+local BasicInfobox = require('Module:Infobox/Basic')
+local Flags = require('Module:Flags')
+
+local Widgets = require('Module:Infobox/Widget/All')
+local Cell = Widgets.Cell
+local Header = Widgets.Header
+local Title = Widgets.Title
+local Center = Widgets.Center
+local Customizable = Widgets.Customizable
+
+local Character = Class.new(BasicInfobox)
+
+function Character.run(frame)
+	local character = Character(frame)
+	return character:createInfobox()
+end
+
+function Character:createInfobox()
+	local infobox = self.infobox
+	local args = self.args
+
+	local widgets = {
+		Header{
+			name = self:nameDisplay(args),
+			subHeader = self:subHeader(args),
+			image = args.image,
+			imageDefault = args.default,
+			imageDark = args.imagedark or args.imagedarkmode,
+			imageDefaultDark = args.defaultdark or args.defaultdarkmode,
+			size = args.imagesize,
+		},
+		Center{content = {args.caption}},
+		Title{name = (args.informationType or 'Hero') .. ' Information'},
+		Cell{name = 'Real Name', content = {args.realname}},
+		Customizable{
+			id = 'country',
+			children = {
+				Cell{
+					name = 'Country',
+					content = {
+						self:_createLocation(args.country)
+					}
+				},
+			}
+		},
+		Customizable{
+			id = 'role',
+			children = {
+				Cell{
+					name = 'Role',
+					content = {args.role}
+				},
+			}
+		},
+		Customizable{
+			id = 'class',
+			children = {
+				Cell{
+					name = 'Class',
+					content = {args.class}
+				},
+			}
+		},
+		Customizable{
+			id = 'release',
+			children = {
+				Cell{
+					name = 'Release Date',
+					content = {args.releasedate}
+				},
+			}
+		},
+		Customizable{id = 'custom', children = {}},
+		Center{content = {args.footnotes}},
+	}
+
+	infobox:categories('Characters')
+	infobox:categories(unpack(self:getWikiCategories(args)))
+
+	local builtInfobox = infobox:widgetInjector(self:createWidgetInjector()):build(widgets)
+
+	if Namespace.isMain() then
+		self:setLpdbData(args)
+	end
+
+	return builtInfobox
+end
+
+function Character:_createLocation(location)
+	if location == nil then
+		return ''
+	end
+
+	return Flags.Icon({flag = location, shouldLink = true}) .. '&nbsp;' ..
+		'[[:Category:' .. location .. '|' .. location .. ']]'
+end
+
+function Character:subHeader(args)
+	return args.title
+end
+
+function Character:getWikiCategories(args)
+	return {}
+end
+
+function Character:nameDisplay(args)
+	return args.name
+end
+
+function Character:setLpdbData(args)
+end
```

Comment:
do we want to set a default storage here?

Reply:
From what I can tell there aren't really any Hero, Operator, Character etc. templates that store data in LPDB. If we want to store data with the standardization I can do that, but I left it off for that reason

Reply:
not saying we have to but we should consider it

Reply:
I've added a few fields as default storage. With how varied the infoboxes on the other wikis are, I'm not 100% sure what should and shouldn't be stored by default. The fields I added in will likely change a lot as more customs are rolled out